src/main/java/sb/persistence/dao/ProductDAO.java

```diff
@@ -0,0 +1,105 @@
+package sb.persistence.dao;
+
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+import sb.domain.entity.Product;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class ProductDAO {
+    private final NamedParameterJdbcTemplate namedJdbcTemplate;
+    private BeanPropertyRowMapper<Product> rowMapper;
+
+    public ProductDAO(NamedParameterJdbcTemplate namedJdbcTemplate) {
+        this.namedJdbcTemplate = namedJdbcTemplate;
+        rowMapper = new BeanPropertyRowMapper<>(Product.class);
+    }
+
+    private final String SQL_SELECT_BY_ID = "select id, supplier_id, name " +
+                                            "from products where id = :id";
+
+    private final String SQL_SELECT_ALL = "select id, supplier_id, name " +
```

Comment:
Where is paging?

Reply:
fix: it was in web-server layer, now transform and concentrate in persistence layer